crates/rpc-server/src/registry.rs

```diff
@@ -40,14 +51,27 @@ pub struct Registry {
     generator: Arc<Generator>,
     mem_pool: Arc<MemPool>,
     store: Store,
+    test_mode: TestMode,
+    tests_rpc_impl: Arc<BoxedTestsRPCImpl>,
```

Comment:
IMO we do not need a trait here, we can use TestModeControl type.

Reply:
But this result in cyclic dependency. ```TestModeControl``` needs ```poa``` and ```rpc-client``` in ```block-producer```.